sdk/messaging/azservicebus/receiver.go

```diff
@@ -453,6 +485,25 @@ func (r *Receiver) receiveMessagesImpl(ctx context.Context, maxMessages int, opt
 	return receivedMessages, nil
 }
 
+// receiveFromCache gets any messages that were retrieved after the Receiver was closed
+// It returns an empty slice when the cache has been exhausted.
+func (r *Receiver) receiveFromCache(maxMessages int) []*ReceivedMessage {
+	if len(r.cachedMessages) == 0 {
+		return nil
+	}
+
+	n := min(len(r.cachedMessages), maxMessages)
+
+	receivedMessages := make([]*ReceivedMessage, n)
+
+	for i := range n {
+		receivedMessages[i] = newReceivedMessage(r.cachedMessages[i], nil)
+	}
+
+	r.cachedMessages = r.cachedMessages[n:]
```

Comment:
Are methods on `Receiver` allowed to be concurrently called? If so, I think this can race.

Reply:
Gah, you're right. I actually thought I had that sealed up, but it's possible for receiver.Close() and receiver.ReceiveMessages() to intersect. Let me think...

Reply:
I updated it to use an atomic.Pointer around the slice value. It's only ever _updated_ once, and never mutated afterwards so this should work just fine.